### ISSUE TITLE
azurerm_web_pubsub_hub not provisioning the resource

### ISSUE THREAD

**Issue by wojciech-czyz-asda** [hashicorp/terraform-provider-azurerm#17784]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.0

### AzureRM Provider Version

3.15.1

### Affected Resource(s)/Data Source(s)

azurerm_web_pubsub_hub

### Terraform Configuration Files

```hcl
resource "azurerm_web_pubsub_hub" "example" {
name = "device_production"
web_pubsub_id = data.azuerm_web_pubsub.webpubsub.id
anonymous_connections_enabled = false
}
```

### Debug Output/Panic Output

```shell
Error: ID was missing the `webPubSub` element
##[error]Error: Terraform Plan failed with exit code: 1
```

### Expected Behaviour

azurerm_web_pubsub_hub provisioned

### Actual Behaviour
Terraform throws:
Error: ID was missing the `webPubSub` element

### Steps to Reproduce

1. Run terraform plan with the given resource
```hcl
resource "azurerm_web_pubsub_hub" "example" {
name = "device_production"
web_pubsub_id = data.azuerm_web_pubsub.webpubsub.id
anonymous_connections_enabled = false
}
```

### Important Factoids

It used to work with provider version 3.14.0 and stopped with 3.15.0

**Comment by xiaxyi** [comment]
@wojciech-czyz-asda Thanks for raising the issue, it should be caused by the case sensitive, I'm working on it.

**Comment by xiaxyi** [comment]
@wojciech-czyz-asda you can track the status in the mentioned pr, once the code is released, you can run `terraform apply --refresh-only --auto-approve` to refresh the state

**Comment by wojciech-czyz-asda** [comment]
Thank you for the response @xiaxyi

**Comment by polys** [comment]
I'm getting the same error on `azurerm_web_pubsub` rather than the hub.
The last version of azurerm that worked was 3.13.0 — all subsequent versions fail with the above error.

```
╷
│ Error: ID was missing the `webPubSub` element
│
│   with module.webpubsub.azurerm_web_pubsub.webpubsub,
│   on modules\webpubsub\main.tf line 1, in resource "azurerm_web_pubsub" "webpubsub":
│    1: resource "azurerm_web_pubsub" "webpubsub" {
│
╵
```

**Comment by xiaxyi** [comment]
Thanks @polys for the comment, the fix is included in the mentioned pr. You can track the status in there, feel free to let me know if there is anything needed.

**Comment by xiaxyi** [comment]
@wojciech-czyz-asda, @polys  good day, just would like to double confirm with you to see if you are still seeing the issue as fix was merged.

**Comment by polys** [comment]
It's been working well for me since the fix. Many thanks @xiaxyi

**Comment by wojciech-czyz-asda** [comment]
Thanks! It works for me as well!

From: Polys Georgiou ***@***.***>
Date: Monday, 9 January 2023 at 10:49
To: hashicorp/terraform-provider-azurerm ***@***.***>
Cc: Wojciech Czyz - Contractor ***@***.***>, Mention ***@***.***>
Subject: Re: [hashicorp/terraform-provider-azurerm] azurerm_web_pubsub_hub not provisioning the resource (Issue #17784)

It's been working well for me since the fix. Many thanks @xiaxyi<https://gbr01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fxiaxyi&data=05%7C01%7Cwojciech.czyz%40asda.uk%7Cab23d02435934a65dc3d08daf226c058%7Cb63ee29faaa64d4e82671d6dca9c0e43%7C0%7C0%7C638088545491836459%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=bCnc5YIjr9hQYMbISyywiO02UXxl7oK67Mi3RBMB1bM%3D&reserved=0>

—
Reply to this email directly, view it on GitHub<https://gbr01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fhashicorp%2Fterraform-provider-azurerm%2Fissues%2F17784%23issuecomment-1375348582&data=05%7C01%7Cwojciech.czyz%40asda.uk%7Cab23d02435934a65dc3d08daf226c058%7Cb63ee29faaa64d4e82671d6dca9c0e43%7C0%7C0%7C638088545491836459%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=2VqpylK5YNw0oXYdE518byH2mMEyov%2B7SAsJ%2FgXB0Z4%3D&reserved=0>, or unsubscribe<https://gbr01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAXJJHXSOXROL6KLZFH4TD7TWRPNJFANCNFSM544TJKPA&data=05%7C01%7Cwojciech.czyz%40asda.uk%7Cab23d02435934a65dc3d08daf226c058%7Cb63ee29faaa64d4e82671d6dca9c0e43%7C0%7C0%7C638088545491836459%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Toecff%2BriHaX76J7kyLRuPpGSaPwD1Q37mYXe0FxRYs%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

This e-mail (including any attachments) is private and confidential and may contain privileged material. If you have received this e-mail in error, please notify the sender and delete it (including any attachments) immediately. You must not copy, distribute, disclose or use any of the information in it or any attachments. Telephone calls may be monitored or recorded.

**Comment by rcskosir** [comment]
@wojciech-czyz-asda Thanks for taking the time to submit this issue. It looks like this has been resolved as of #17823. As such, I am going to mark this issue as closed.
